### PR TITLE
MGDCTRS-1566 Stop creating new/duplicated conditions in the ManagedConnector resource with every reconciliation run

### PR DESCRIPTION
**Pull request by rk3rn3r**
closes https://issues.redhat.com/browse/MGDCTRS-1566

### CODE REVIEW

File: cos-fleetshard-operator-debezium/src/main/java/org/bf2/cos/fleetshard/operator/debezium/model/ConnectorStatus.java [hunk 1]
```diff
@@ -22,12 +26,50 @@ public ConnectorStatusSpec getStatusSpec() {
...
     public List<Condition> getStatusSpecConditions() {
+        return new ArrayList<>(getConditions().values());
+    }
+
+    private Map<String, Condition> getConditions() {
         if (null == statusSpecConditions) {
-            statusSpecConditions = statusSpec.getConditions();
+            statusSpecConditions = new HashMap<>(statusSpec.getConditions().size());
+            for (var condition : statusSpec.getConditions()) {
+                statusSpecConditions.put(condition.getType(), condition);
```
Comment (rk3rn3r) [inline cos-fleetshard-operator-debezium/src/main/java/org/bf2/cos/fleetshard/operator/debezium/model/ConnectorStatus.java:36]: Conditions are now organized as a map Type => $Condition_Object.

File: cos-fleetshard-operator-debezium/src/main/java/org/bf2/cos/fleetshard/operator/debezium/model/ConnectorStatus.java [hunk 2]
```diff
@@ -22,12 +26,50 @@ public ConnectorStatusSpec getStatusSpec() {
...
+        var lastCondition = getConditions().get(condition.getType());
+        if (null == lastCondition) {
+            return false;
+        }
+        return isSimilarCondition(lastCondition, condition);
+    }
+
+    public void setCondition(Condition condition) {
+        if (!hasSimilarCondition(condition)) {
+            getConditions().put(condition.getType(), condition);
```
Comment (rk3rn3r) [inline cos-fleetshard-operator-debezium/src/main/java/org/bf2/cos/fleetshard/operator/debezium/model/ConnectorStatus.java:69]: Only add or replace when the condition content is new. Else we keep the previous condition/s.

**Comment by rk3rn3r** [comment]
TODO: remove old conditions that don't apply anymore.

**Review by lburgazzoli — APPROVED** [review]
Can you add a test as a follow up ?

**Comment by rk3rn3r** [comment]
@lburgazzoli I filed https://issues.redhat.com/browse/MGDCTRS-1602 for this.

**Pull request merged**